packages/transformer/src/IModelTransformer.ts

```diff
@@ -460,6 +460,47 @@ export class IModelTransformer extends IModelExportHandler {
     return aspectProps;
   }
 
+  public static initRelationshipProvenanceOptions(
+    sourceRelInstanceId: Id64String, 
+    targetRelInstanceId: Id64String,
+    args: {
+      sourceDb: IModelDb;
+      targetDb: IModelDb;
+      isReverseSynchronization: boolean;
+      targetScopeElementId: Id64String;
+      forceOldRelationshipProvenanceMethod: boolean;
+    },
+  ): ExternalSourceAspectProps {
+    function  checkReverseSynchronization(source: any, target: any): any {
```

Comment:
I wouldn't abstract this. We already have a concept of `provenanceDb` and `provenanceSourceDb` (naming is confusing) for this, just do this:

```ts
// instead of dbToUse
const provenanceDb = args.isReverseSynchronization ? source : target;
```

Also almost never use `any`. This loses all type checking on `dbToUse` later which can be a large issue for maintenance.